flambe/nlp/common/dataset_tabular_compressed.py

```diff
@@ -0,0 +1,93 @@
+from typing import Dict, Union
+
+import abc
+import os
+import tempfile
+
+from urllib.parse import unquote, urlparse
+import requests
+import shutil
+
+from flambe.dataset import TabularDataset
+from flambe.field import Field
+
+
+class CompressedDataTabularDataset(TabularDataset, metaclass=abc.ABCMeta):
```

Comment:
I think it makes more sense to bake this functionality into the TabularDataset so everyone benefits from it